File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/factories/primitive/CylinderFactory.kt
package io.micronaut.docs.factories.primitive

// tag::imports[]
import io.micronaut.context.annotation.Bean
import io.micronaut.context.annotation.Factory
import jakarta.inject.Named
// end::imports[]

// tag::class[]
@Factory
class CylinderFactory {
    @get:Bean
    @get:Named("V8") // <1>
    val v8 = 8

    @get:Bean
    @get:Named("V6") // <1>
    val v6 = 6
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/factories/primitive/CylinderFactory.py
# tag::imports[]
from micronaut.context.annotation import Bean, Factory
from jakarta.inject import Named
from typing import Annotated
# end::imports[]

# tag::class[]
@Factory
class CylinderFactory:
    v8 : Annotated[int, Bean, Named("V8")] = 8 # (1)
    v6 : Annotated[int, Bean, Named("V6")] = 6 # (1)
# end::class[]
